tokio/src/runtime/io/registration_set.rs

```diff
@@ -123,7 +123,7 @@ unsafe impl linked_list::Link for Arc<ScheduledIo> {
 
     unsafe fn from_raw(ptr: NonNull<Self::Target>) -> Arc<ScheduledIo> {
         // safety: the linked list currently owns a ref count
-        unsafe { Arc::from_raw(ptr.as_ptr() as *const _) }
+        Arc::from_raw(ptr.as_ptr() as *const _)
     }
```

Comment:
This is not a change I want. Rather, I agree with [the unsafe-block-in-unsafe-fn RFC][1]. I would rather enable `#[warn(unsafe_op_in_unsafe_fn)]` and fix anywhere that is missing an unsafe block inside unsafe functions.

[1]: https://rust-lang.github.io/rfcs/2585-unsafe-block-in-unsafe-fn.html